daisy_workflows/export_metadata/export-metadata.ps1

```diff
@@ -0,0 +1,87 @@
+function Get-MetadataValue {
+    <#
+    .SYNOPSIS
+      Returns a value for a given metadata key.
+
+    .PARAMETER $key
+      The metadata key to retrieve.
+
+    .PARAMETER $default
+      The value to return if the key is not found.
+
+    .RETURNS
+      The value for the key or null.
+  #>
+    param (
+        [parameter(Mandatory=$true)]
+        [string]$key,
+        [parameter(Mandatory=$false)]
+        [string]$default
+    )
+
+    # Returns the provided metadata value for a given key.
+    $url = "http://metadata.google.internal/computeMetadata/v1/instance/attributes/$key"
+    try {
+        $client = New-Object Net.WebClient
+        $client.Headers.Add('Metadata-Flavor', 'Google')
+        return ($client.DownloadString($url)).Trim()
+    }
+    catch [System.Net.WebException] {
+        if ($default) {
+            return $default
+        }
+        else {
+            Write-Host "Failed to retrieve value for $key."
+            return $null
+        }
+    }
+}
+
+function Export-ImageMetadata {
+    $image_version = Get-Date -Format "yyyyMMdd"
+    $build_date = Get-Date -Format "o"
```

Comment:
$image_version feels like it should be a value that is passed into the workflow as it's contained in the disk metadata.

Instead of $build_date, I'd suggest this should be metadata_date as this isn't when the image is built it's when the image is having metadata exported.

Reply:
since the metadata is help for us auditing all the publish image, the $image_version we want here is more about the yyyyMMdd suffix we used in tar.gz and image name instead the real image version from OS itself (we are less care about). for build_date, we have a borg job running 5 mins periodically to monitor any new published image and we just assume that we always publish immediately after build.

Reply:
The Windows builds have a 1-2 day lag between build and publish. So if they're build on 20210222, they'll be published on 20210224 if all goes well. From your intent, it sounds like we'd be better off passing in the value to ensure it's accurate instead of derive it from the current datetime. 

This would especially be true for the derived builds as they might be build the day after the OS builds, but they'd still have our build date of v20210222.